src/fr/phenixscans/src/eu/kanade/tachiyomi/extension/fr/phenixscans/PhenixScans.kt

```diff
@@ -1,24 +1,304 @@
+
+            Filter.Header("Sort by"),
+            SortFilter(),
+            Filter.Separator(),
+
+            Filter.Header("Status"),
+            StatusFilter(),
+            Filter.Separator(),
+        )
+
+        if (filtersState == FiltersState.FETCHED) {
+            filters += listOf(
+                Filter.Separator(),
+                Filter.Header("Filter by genres"),
+                GenreFilter(genresList),
+            )
+        } else {
+            filters += listOf(
+                Filter.Separator(),
+                Filter.Header("Click on 'Reset' to load missing filters"),
+            )
+        }
+
+        return FilterList(filters)
+    }
+
+    private var genresList: List<Tag> = emptyList()
+    private var fetchFiltersAttempts = 0
+    private var filtersState = FiltersState.NOT_FETCHED
+
+    private fun fetchFilters() {
+        if (filtersState != FiltersState.NOT_FETCHED || fetchFiltersAttempts >= 3) return
+        filtersState = FiltersState.FETCHING
+        fetchFiltersAttempts++
+        thread {
+            try {
+                val response = client.newCall(GET("$apiBaseUrl/genres", headers)).execute()
+                val filters = response.parseAs<GenresDto>()
+
+                genresList = filters.data.map { Tag(it.name, it.id) }
+
+                filtersState = FiltersState.FETCHED
+            } catch (e: Throwable) {
+                filtersState = FiltersState.NOT_FETCHED
+            }
+        }
+    }
+
+    open class UriPartFilter(displayName: String, val vals: Array<Pair<String, String>>) :
+        Filter.Select<String>(displayName, vals.map { it.first }.toTypedArray()) {
+        fun toUriPart() = vals[state].second
+    }
+
+    private enum class FiltersState { NOT_FETCHED, FETCHING, FETCHED }
+
+    // =============================== Manga ==================================
+
+    override fun mangaDetailsRequest(manga: SManga): Request {
+        val url = apiBaseUrl.toHttpUrl().newBuilder()
+            .addPathSegment("front").addPathSegment("manga")
+            .addPathSegment(manga.url.substringAfterLast("manga/"))
+            .build()
+
+        Log.e("PhenixScans", url.toString())
+
+        return GET(url.toString(), headers)
+    }
+
+    override fun mangaDetailsParse(response: Response): SManga {
+        val data = response.parseAs<MangaDetailResponse>()
+
+        return SManga.create().apply {
+            title = data.manga.title
+            thumbnail_url = "$apiBaseUrl/${data.manga.coverImage}"
+            url = "/manga/${data.manga.slug}"
+            description = data.manga.synopsis
+        }
+    }
+
+    // ============================== Chapters ==============================
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val data = response.parseAs<MangaDetailResponse>()
+
+        return data.chapters.map {
+            SChapter.create().apply {
+                chapter_number = it.number.float
```

Comment:
Do they have special chapters? Do these work here?

Reply:
I checked the full db, the only thing I found was a manga not published but available by search. Every chapter_number is a float or int.